app/src/main/java/com/mstudio/superheroarch/presentation/overview/MainActivity.kt

```diff
@@ -77,10 +86,14 @@ class MainActivity : AppCompatActivity(), MainViewTranslator {
         adapter.submitList(characters)
     }
 
-    override fun showEmptyCharactersError() {
+    override fun showEmptyCharactersError(isFavoriteSection: Boolean) {
         binding.charactersRv.visibility = View.GONE
         binding.errorBody.visibility = View.VISIBLE
-        binding.errorBody.text = resources.getString(R.string.main_title_empty_body)
+        binding.errorBody.text = if (isFavoriteSection) {
```

Comment:
another way to handle this would be to put a generic message, something like "there are no characters matching this filter" and then you won't need the boolean here.